File: ohjelmointikielet/zoo-java/Tiger.java

package fi.lab.zoo;

/**
 * @author Anni Karja
 */
public class Tiger extends Animal {
    private String species;
    
    public Tiger() {
        this("Leevi", new Date(), 100, "Siberian");
    }
    
    public Tiger(String name, Date birth, double weight, String species) {
        super(name, birth, weight);
        setSpecies(species);
    }
    @Override
    public String toString() {
        return super.toString() + " Species: " + getSpecies();
    }
    @Override
    public void move() {
        System.out.println("Tiger " + getName() + " has moved!");
    }
    @Override
    public void utter() {
        System.out.println("Tiger " + getName() + " goes grr!");
    }

    /**
     * @return the species
     */
    public String getSpecies() {
        return species;
    }

    /**
     * @param species the species to set
     */
    public void setSpecies(String species) {
        this.species = species;
    }
}

File: ohjelmointikielet/zoo-cs/Tiger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Zoo
{
    class Tiger : Animal
    {
        private string species;

        public string Species { get => species; set => species = value; }

        public Tiger(string name = "Leevi", Date birth = null, double weight = 100, string species = "Siberian")
            : base(name, birth, weight)
        {
            Species = species;
        }
        public override void move()
        {
            Console.WriteLine($"Tiger {Name} has moved!");
        }

        public override void utter()
        {
            Console.WriteLine($"Tiger {Name} goes grr!");
        }

        public override string ToString()
        {
            return $"{base.ToString()} Species: {Species}";
        }

    }
}
